x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +871,196 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[1], emissionRateForExternalTokens, externalIncentiveClPoolEpoch2[1]) // new record
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), internalGaugeDecCoins[0], emissionRateForInternalTokens, externalIncentiveClPoolEpoch2[2])       // new record
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[0], emissionRateForExternalTokens, externalIncentiveClPoolEpoch2[3]) // old record
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[1], emissionRateForExternalTokens, externalIncentiveClPoolEpoch2[4]) // old record
+
+	// 8. let epoch 3 pass
+	// ******************** EPOCH 3 *********************
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.Identifier, 3)
+
+	s.ValidateDistributedGauge(clPoolInternalGaugeId, 3, internalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 2, externalGaugeCoins)
+
+	internalIncentiveClPoolEpoch3, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPoolEpoch3, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	s.Require().Equal(internalIncentiveClPoolEpoch2, internalIncentiveClPoolEpoch3)
+	s.Require().Equal(externalIncentiveClPoolEpoch2, externalIncentiveClPoolEpoch3)
+}
+
+func (s *KeeperTestSuite) CreateNoLockExternalGauges(clPoolId1, clPoolId2 uint64, externalGaugeCoins sdk.Coins) ([]uint64, uint64, uint64) {
+	var gaugeIdWithDuration []uint64
+
+	// Create 1 external no-lock gauge perpetual over 1 epochs MsgCreateGauge
+	clPoolExternalGaugeId, err := s.App.IncentivesKeeper.CreateGauge(s.Ctx, true, s.TestAccs[1], externalGaugeCoins,
+		lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.NoLock,
+		},
+		s.Ctx.BlockTime(),
+		1,
+		clPoolId1,
+	)
+	s.Require().NoError(err)
+
+	// Create 1 external no-lock gauge non-perpetual over 2 epochs MsgCreateGauge
+	clPoolExternalGaugeId2, err := s.App.IncentivesKeeper.CreateGauge(s.Ctx, false, s.TestAccs[2], externalGaugeCoins,
+		lockuptypes.QueryCondition{
+			LockQueryType: lockuptypes.NoLock,
+		},
+		s.Ctx.BlockTime(),
+		2,
+		clPoolId2,
+	)
+	s.Require().NoError(err)
+
+	return gaugeIdWithDuration, clPoolExternalGaugeId, clPoolExternalGaugeId2
+}
+
+func (s *KeeperTestSuite) IncentivizeInternalGauge(poolId1, poolId2 uint64, epochDuration time.Duration, removeDistrRecord bool) {
+	var weight sdk.Int
+	if !removeDistrRecord {
+		weight = sdk.NewInt(100)
+	} else {
+		weight = sdk.ZeroInt()
+	}
+
+	getGaugeIdForPool1, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, poolId1, epochDuration)
+	s.Require().NoError(err)
+
+	getGaugeIdForPool2, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, poolId2, epochDuration)
+	s.Require().NoError(err)
```

Comment:
Then, iterate over that slice and create the pool <> gauge id link + append a distr record inside the loop.

This would help reduce code duplication